proxy/hdrs/MIME.cc

```diff
@@ -2569,8 +2569,8 @@ mime_parser_parse(MIMEParser *parser, HdrHeap *heap, MIMEHdrImpl *mh, const char
     //////////////////////////////////////////////////////////////////////
 
     if (must_copy_strings || (!line_is_real)) {
-      char *dup      = heap->duplicate_str(parsed.data(), parsed.size());
-      intptr_t delta = dup - parsed.data();
+      char *dup  = heap->duplicate_str(parsed.data(), parsed.size());
+      auto delta = dup - parsed.data();
```

Comment:
We shouldn't use auto in the code in situations like this, it detailed information from the code making it harder to read/understand.

Reply:
Do you need to know the type? In any case, `intptr_t` was the wrong type, it should be `ptrdiff_t`. The use of `auto` means the compiler will pick the best type.

I've been shifting my usage because in many cases like this the compiler will actually warn you to use `auto`, e.g.
```
Thing * thing = new Thing; // Compiler says to use 'auto' here.
```

Reply:
It makes sense to use auto when using new, since the type is already on the line.
Doing things like auto `item = foo.get(); ` doesn't make sense and makes the code harder to read.